src/components/dailyQuestsPage/DailyQuests.jsx

```diff
@@ -0,0 +1,244 @@
+import SuggestedQuests from "../dailyQuestsPage/SuggestedQuests";
+import WeekDaysCarousel from "./WeekDaysCarousel";
+import QuestList from "./QuestList";
+import AddQuestButton from "./AddQuestButton";
+
+const daysOfWeek = ["SUN", "MON", "TUE", "WED", "THU", "FRI", "SAT"];
+
+const normalizeFrequency = (freq) => {
+  if (!freq) return "";
+  const array = Array.isArray(freq)
+    ? freq
+    : freq.split(",").map((d) => d.trim());
+  return array.sort().join(", ");
+};
+
+const DailyQuests = () => {
+  const [currentDate, setCurrentDate] = useState(new Date());
+  const [selectedDay, setSelectedDay] = useState(currentDate);
+  const [quests, setQuests] = useState([]);
+  const [loading, setLoading] = useState(true);
+
+  const [showModal, setShowModal] = useState(false);
+  const [selectedQuest, setSelectedQuest] = useState(null);
+  const [title, setTitle] = useState("");
+  const [frequency, setFrequency] = useState("Daily");
+  const [icon, setIcon] = useState("+");
+
+  useEffect(() => {
+    const savedBookQuests =
+      JSON.parse(localStorage.getItem("readBooksQuests")) || [];
+
+    const defaultQuests = [
+      {
+        id: 1,
+        title: "Drink water",
+        icon: "💧",
+        frequency: "Daily",
+        completed: {},
+      },
+      {
+        id: 2,
+        title: "Stretch",
+        icon: "🤸",
+        frequency: "Mondays, Wednesdays",
+        completed: {},
+      },
+      {
+        id: 3,
+        title: "Read 1 page",
+        icon: "📘",
+        frequency: "Daily",
+        completed: {},
+      },
+      {
+        id: 4,
+        title: "Meditate for 10 minutes",
+        icon: "✨",
+        frequency: "Daily",
+        completed: {},
+      },
+      {
+        id: 5,
+        title: "Go for a walk",
+        icon: "🚶‍♂️",
+        frequency: "Fridays",
+        completed: {},
+      },
+      {
+        id: 6,
+        title: "Sleep early",
+        icon: "🛏",
+        frequency: "Mondays, Wednesdays, Fridays",
+        completed: {},
+      },
+      {
+        id: 7,
+        title: "Write journal entry",
+        icon: "✍️",
+        frequency: "Fridays",
+        completed: {},
+      },
+    ];
+
+    const filteredDefaultQuests = defaultQuests.filter((def) => {
+      return !savedBookQuests.some((saved) => {
+        const savedFreq = normalizeFrequency(saved.frequency);
+        const defFreq = normalizeFrequency(def.frequency);
+        return (
+          saved.title === def.title &&
+          saved.icon === def.icon &&
+          savedFreq === defFreq
+        );
+      });
+    });
+
+    const combinedQuests = [
+      ...savedBookQuests.map((q) => ({ ...q, completed: {} })),
+      ...filteredDefaultQuests,
+    ];
+
+    setQuests(combinedQuests);
+    setLoading(false);
+  }, []);
+
+  const handleToggle = (questId) => {
+    const dateKey = format(selectedDay, "yyyy-MM-dd");
+    setQuests((prev) =>
+      prev.map((q) =>
```

Comment:
```suggestion
      prev.map((quest) => 
```

will make code more readable